Inc/ResourceUploadBatch.h

```diff
@@ -41,9 +43,14 @@ namespace DirectX
         // The resource must be in the COPY_DEST state.
         void __cdecl Upload(
             _In_ ID3D12Resource* resource,
-            _In_ uint32_t subresourceIndexStart,
+            uint32_t subresourceIndexStart,
             _In_reads_(numSubresources) D3D12_SUBRESOURCE_DATA* subRes,
-            _In_ uint32_t numSubresources);
+            uint32_t numSubresources);
+
+        void __cdecl Upload(
```

Comment:
To support this, I modified ``ResourceUploadBatch`` to have an overload that will take data from a ``SharedGraphicsResource``.